nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/HazelcastCore.java

```diff
@@ -411,6 +411,9 @@ private void setPayaraSerializerConfig(SerializationConfig serConfig) {
 
     private void buildNetworkConfiguration(Config config) throws NumberFormatException {
         NetworkConfig nConfig = config.getNetworkConfig();
+        String noClusterProp = System.getProperty("payaramicro.noCluster");
+        if (noClusterProp != null && Boolean.parseBoolean(noClusterProp))
+            config.getNetworkConfig().getJoin().getAutoDetectionConfig().setEnabled(false);
```

Comment:
I wouldn't do this as a system property - this strikes me as something that should be persisted to the Hazelcast config in the domain.xml.

You should add an attribute to the `HazelcastConfigSpecificConfiguration` class:
```
    @Attribute(defaultValue = "true", dataType = Boolean.class)
    String getClusteringEnabled();
    public void setClusteringEnabled(String value);

```

...and then use it:

```suggestion
        String noClusterProp = nodeConfig.getClusteringEnabled();
        if (noClusterProp != null && Boolean.parseBoolean(noClusterProp)) {
            config.getNetworkConfig().getJoin().getAutoDetectionConfig().setEnabled(false);
        }
```

Also we try to avoid unbracketed `if` statements here :)